### PR TITLE
[rn-randomnumber] Added definitions

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/rn-randomnumber/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for rn-randomnumber 1.0
+// Project: https://www.npmjs.com/package/rn-randomnumber
+// Definitions by: Rajas Paranjpe <https://github.com/ChocolateLoverRaj>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function randomNoGenerator(min: number, max: number): number;
```
Comment (peterblazejewicz) [inline types/rn-randomnumber/index.d.ts:6]:
```suggestion
declare function randomNoGenerator(min?: number, max?: number): number;
```
it support quick and dirty usage:
```ts
> rn()
0.7837907906151589
> rn()
0.61723048281547
> rn()
0.650512321255567
```

File: types/rn-randomnumber/rn-randomnumber-tests.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import random = require('rn-randomnumber');
+
+random(-100, 25); // $ExpectType number
+
+random(8); // $ExpectError
```
Comment (peterblazejewicz) [inline types/rn-randomnumber/rn-randomnumber-tests.ts:5]:
```ts
random(8); // $ExpectError
```
that's not true:
```ts
typeof(rn(8))
'number'
```

Comment (ChocolateLoverRaj) [reply types/rn-randomnumber/rn-randomnumber-tests.ts:5]: @peterblazejewicz I will fix that.

The documentation and [code](https://cdn.jsdelivr.net/npm/rn-randomnumber@1.0.2/index.js) don't make it clear that it works with just one param. The code relies on `undedined - number`, which then returns a number within `[0, n}`.

Comment (peterblazejewicz) [reply types/rn-randomnumber/rn-randomnumber-tests.ts:5]: please, run the code, it works the way I've described. There is no docs for this. I believe this is how rand impl works, give me a rand number (whatever) or let me give a range to pick from.
If you stick with NaN ($ExpectError), the method stamp should be ovverrides:
```ts
declare function randomNoGenerator(min: number, max: number): number;
declare function randomNoGenerator(): number;
```
(give me one from a range, or just give me whatever)

Comment (ChocolateLoverRaj) [reply types/rn-randomnumber/rn-randomnumber-tests.ts:5]: @peterblazejewicz as you said `rn(8)` works, so I'll leave it as `declare function randomNoGenerator(min?: number, max?: number): number;`

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ChocolateLoverRaj when published, please play with the code.
If you feel, this single optional param should be disallowed, just update to no-param or 2-required-params version as discussed, thx!

**Comment by ChocolateLoverRaj** [comment]
> LGTM! @ChocolateLoverRaj when published, please play with the code. If you feel, this single optional param should be disallowed, just update to no-param or 2-required-params version as discussed, thx!

@peterblazejewicz  I played with the code before merging. Since `NaN` is not useful, I will remove single param usage and do the overloads as you suggested.

**Review by peterblazejewicz — APPROVED** [review]

**Comment by ChocolateLoverRaj** [comment]
Ready to merge

**Pull request merged**